ext/canvas/lib.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::error::type_error;
+use deno_core::error::AnyError;
+use deno_core::op2;
+use deno_core::ToJsBuffer;
+use image::imageops::FilterType;
+use image::ColorType;
+use image::ImageDecoder;
+use image::Pixel;
+use image::RgbaImage;
+use serde::Deserialize;
+use serde::Serialize;
+use std::path::PathBuf;
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+enum ImageResizeQuality {
+  Pixelated,
+  Low,
+  Medium,
+  High,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct ImageProcessArgs {
+  width: u32,
+  height: u32,
+  surface_width: u32,
+  surface_height: u32,
+  input_x: i64,
+  input_y: i64,
+  output_width: u32,
+  output_height: u32,
+  resize_quality: ImageResizeQuality,
+  flip_y: bool,
+  premultiply: Option<bool>,
+}
+
+#[op2]
+#[serde]
+fn op_image_process(
+  #[buffer] buf: &[u8],
+  #[serde] args: ImageProcessArgs,
+) -> Result<ToJsBuffer, AnyError> {
+  let view =
+    RgbaImage::from_vec(args.width, args.height, buf.to_vec()).unwrap();
+  let mut surface = RgbaImage::new(args.surface_width, args.surface_height);
+
+  image::imageops::overlay(&mut surface, &view, args.input_x, args.input_y);
```

Comment:
We should add a fast path which does not allocate this extra surface when surface dimenstions == view dimenstion (which I assume is always the case unless source rectangle is specified from the API).

Can you add a TODO for this?

Reply:
I implemented it, PTAL